bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdMetadataDefaults.java

```diff
@@ -13,21 +13,29 @@
  *******************************************************************************/
 package org.eclipse.cdt.lsp.internal.clangd;
 
-import java.util.Optional;
+import java.util.stream.Collectors;
 
 import org.eclipse.cdt.lsp.clangd.ClangdMetadata;
+import org.eclipse.cdt.lsp.clangd.ClangdOptions;
 import org.eclipse.cdt.lsp.internal.clangd.editor.LspEditorUiMessages;
-import org.eclipse.cdt.utils.PathUtil;
-import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.preferences.PreferenceMetadata;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
-final class ClangdMetadataDefaults implements ClangdMetadata {
+@Component
+public final class ClangdMetadataDefaults implements ClangdMetadata {
+
+	@Reference
+	private ClangdOptions defaults;
```

Comment:
Let's declare another interface to fill defaults.
`ClangdMetadataDefaults` is to define preferences metafata, including default value.While `ClangdOptions` it to calculate actual option values using metadata (and defaults)
So, reusing `ClangdOptions` to define default option values creates a kind of circularity.

Reply:
I'd call the new interface `ClangdOptionValues` and move all methods except `toList` from `ClangdOptions` there. The question is if `ClangdOptions` extends `ClangdOptionValues`, I think it shouldn't, wdyt @ruspl-afed?

Reply:
Thinking about it again, the simplest solution would be to remove `toList` from ClangdOptions and move it to ClangdConfiguration. This makes the most sense to me, because the available options are one thing and the list of commands for clangd is another thing. I will add another commit, maybe you can have a look if you agree.

Reply:
See 2c8d30862d9881a5e957d05093a21a5896872ab8

Reply:
> would be to remove toList from ClangdOptions and move it to ClangdConfiguration. 

And what `toList` will return for `ClangdConfiguration` in general? It needs context first to resolve `ClangdOptions` with values and then represent it as a list.

Reply:
> > would be to remove toList from ClangdOptions and move it to ClangdConfiguration.
> 
> And what `toList` will return for `ClangdConfiguration` in general? It needs context first to resolve `ClangdOptions` with values and then represent it as a list.

That's true, it needs the context, just like the options method does.

Reply:
See 34455cfcfa0a850a17ab5f663919e94e034d6dcf